testing.py

```diff
@@ -1372,10 +1382,12 @@ def json(self):
         content = self.content
         # The queued content might be a JSON string or it might
         # just be the object you'd get from loading a JSON string.
-        if isinstance(content, basestring):
+        if isinstance(content, (unicode, bytes)):
```

Comment:
`2to3` replaces 'basestring' with 'str', which means Unicode strings as opposed to bytestrings. This is how we say 'both types of strings' in a way that will convert over.

Reply:
According to the porting guide, `2to3` handles this in a more sensible way, but it wasn't working here for some reason.